core/common/connector-core/src/main/java/org/eclipse/edc/connector/core/security/LocalPublicKeyServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.core.security;
+
+import org.eclipse.edc.spi.iam.LocalPublicKeyService;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.spi.security.KeyParserRegistry;
+import org.eclipse.edc.spi.security.Vault;
+
+import java.security.PublicKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link LocalPublicKeyService} which looks-up for the key by id
+ * first in the locally cached keys and as fallback in the vault.
+ */
+public class LocalPublicKeyServiceImpl implements LocalPublicKeyService {
```

Comment:
Contrary to the PR Description, the local service does not resolve keys from the config (via path or raw string). I think we will need that though.

Reply:
It resolve from the config actually, just it sees the public keys from the config and cached locally

Reply:
The path is not supported yet, but do we need it?

Reply:
putting a raw PEM string as config value may be cumbersome, and may cause issues with multiline etc., so being able reference a key file (`.pem`) may come in handy

Reply:
I can add another field to the config which is `path` instead of `value` and read it from a path in case

Reply:
i would do both: check if a `*.path` config entry was specified, and if not, check if a `.value` config entry is there. That would give: vault -> path -> raw value

Reply:
done